core/src/test/java/org/web3j/protocol/websocket/WebSocketServiceTest.java

```diff
@@ -121,6 +122,21 @@ public void testThrowExceptionIfConnectionFailed() throws Exception {
                 });
     }
 
+    @Test
+    public void testReconnectIfFirstConnectionFailed() throws Exception {
+        when(webSocketClient.connectBlocking()).thenReturn(false);
+        assertThrows(
+                ConnectException.class,
+                () -> {
+                    service.connect();
+                });
+        service.connect();
+        // reconnectBlocking() should be called if 1st attempt was failed, if we call
+        // connectBlocking() for a second time, we will get IllegalStateException with real
+        // webSocketClient
+        verify(webSocketClient, times(1)).reconnectBlocking();
```

Comment:
```suggestion
        verify(webSocketClient, atMostOnce()).reconnectBlocking();
```
nit, but to be consistent with other test cases

Reply:
I prefer times too, but also I like classes to be as clear and consistent as possible, so adding that now would mean to update all checks in test and is not really the case. Thanks for update!